### ISSUE TITLE
Option to only play gifs on hover

### ISSUE THREAD

**Issue by geekgonecrazy** [RocketChat/Rocket.Chat#4055]
On native clients we seem to run into a problem where too many gifs playing causes the render loop to get blocked.  We need to adopt what other platforms have done, and don't play them automatically.  Maybe only hover?

**Comment by Jellyfrog** [comment]
or play X times and then switch to on hover

**Comment by FibreFoX** [comment]
What gifs do you mean? Is there some native support for providers like imgur? I just know that https://github.com/jeffreytierney/giphbot webhook?

**Comment by geekgonecrazy** [comment]
@Jellyfrog that would work too.

@FibreFoX so if you post a link to a gif it'll show up and start playing.  These might be posted by a user, by gifRocket etc.  The idea is to not have those play automatically.  Just imagine someone went a little gif crazy and pasted links to 10-15 gifs.  You now have all 10-15 of those gifs playing at the same time.  Eating through resources that you don't necessarily want them to be.  It can bog down your chat.

**Comment by subesokun** [comment]
I'd love to see a play button just like Twitter does. Once the animation has completed it should stay in paused mode. Also it should remember that the Gif has been played already once so that it won't be played again if you enter the room or scroll around.

**Comment by FibreFoX** [comment]
This might be a customizable option of the user, but not "on hover", mobile devices don't have hover, "on click"/"on tap" would be nice to have available here.

**Comment by gabrielsixel** [comment]
Something like this: http://rubentd.com/gifplayer/ ?

**Comment by geekgonecrazy** [comment]
@gabrielsixel exactly.  Something that wouldn't bog down the render thread.

**Comment by Ashray123** [comment]
I want to work on this issue, please someone point me to the file in which the changes are to be made .

**Comment by sushant52** [comment]
@geekgonecrazy @ggazzo We could convert the GIF to .mp4 video format and use the video controls to play/pause the GIF, and also add a loop, autoplay, muted options to give it more GIFy feel if one wants to. 

## Examples of popular website already doing this:-
-  [Twitter converts animated GIFs to MP4 files on upload](https://mashable.com/2014/06/20/twitter-gifs-mp4/#Mtz26DX1BZqG)
-  [Pinterest improves performance by converting GIF to mp4](https://medium.com/pinterest-engineering/improving-gif-performance-on-pinterest-8dad74bf92f1)
-  [IMGUR, a site full of GIFs, converts GIFs to HTML5 video](https://blog.imgur.com/2014/10/09/introducing-gifv/)

This not only provides more controls with simple already built controls, but also reduces **file size** and improves performance of servers.
[**Source**](https://www.smashingmagazine.com/2018/11/gif-to-video/)
What say?

**Comment by ritwizsinha** [comment]
@geekgonecrazy I am working on this feature, I will first try to implement a simple click/tap for play and stop and if that looks good may be I could do some of the optimizations mentioned above. Thanks

**Comment by ritwizsinha** [comment]
The image container is common for both gifs and normal images, maybe I would have to create new container for gifs

**Comment by geekgonecrazy** [comment]
@RocketChat/frontend if work here won't be accepted please chime in so no one wastes time 🙏

**Comment by ritwizsinha** [comment]
@geekgonecrazy I looked at the code and understood, what had to be done
![gif-issue-rocketchat](https://user-images.githubusercontent.com/43509699/115888096-2771e480-a470-11eb-82a2-a696f28abd0b.gif).
I tried adding a wrapper for Image component and a variable controlling toggle when clicked
One problem arises is how to care of the open attachment feature when clicked, when the gif is running, clicking on the gif does two things as shown above (opens the attachment  and stops the gif).
One solution is to remove the action of stopping a gif and automatically stopping it after one run. 
wdyt?

**Comment by MondaySam** [comment]
Please make the gif stop playing after one run.
We like to post reaction gifs via /giphy but it quickly gets annoying when those gifs are constantly running.
Especially when you have the chat open on another screen and constantly see movement in your peripheral view.

**Comment by 3id0** [comment]
> Please make the gif stop playing after one run or two. We like to post reaction gifs via /giphy but it quickly gets annoying when those gifs are constantly running. Especially when you have the chat open on another screen and constantly see movement in your peripheral view.

I disagree. I mean its a sane option as a default but probably better to let the user choose with a configuration option on the client side.
Also you are allowed to minimize your chat window if it bothers you :p

**Comment by sennewood** [comment]
GIFs need autoplay.

**Comment by norogoth** [comment]
@sennewood is correct. If GIFs in Rocketchat can't autoplay, is life even truly worth living?

**Comment by stephentgrammer** [comment]
It would be amazing if we could revisit this issue. It would be wonderful to be able to pause endlessly playing gifs.
